experimental/devices/sn3218/sn3218_test.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package sn3218
+
+import (
+	"bytes"
+	"testing"
+
+	"periph.io/x/periph/conn/i2c/i2ctest"
+)
+
+func setup() *i2ctest.Record {
+	bus := i2ctest.Record{
+		Bus: nil,
+		Ops: []i2ctest.IO{},
+	}
+	return &bus
+}
+
+func TestNew(t *testing.T) {
+	bus := i2ctest.Record{
+		Bus: nil,
+		Ops: []i2ctest.IO{},
+	}
+	_, err := New(&bus)
+	if err != nil {
+		t.Fatal("New should not return error", err)
+	}
+	if len(bus.Ops) > 1 {
+		t.Fatal("Expected 0 operation to I2CBus, got ", len(bus.Ops))
+	}
+
+	if !bytes.Equal(bus.Ops[0].W, []byte{0x17, 0xFF}) {
+		t.Fatal("Expected: 0x17, 0xFF (reset), got: ", bus.Ops[0].W)
+	}
+}
+
+func TestHalt(t *testing.T) {
+	bus := setup()
+	dev, _ := New(bus)
+	err := dev.Halt()
```

Comment:
Inline into next line.

Reply:
you mean just add an empty line? Then done. Otherwise, please explain

Reply:
if err := dev.Halt(); err != nil {

Reply:
fixed, will send another pull requests together with some more improvements.